pkg/server/message.go

```diff
@@ -254,64 +282,101 @@ func (t *TranslateServiceServer) UpdateMessages(
 
 // helpers
 
-// populateTranslatedMessages adds any missing messages from the original messages to all translated messages.
-func (t *TranslateServiceServer) populateTranslatedMessages(
+// alterTranslations replaces text in messages for all translations
+// based on changes in the original messages text, returns messagesSlice containing altered translations.
+func (t *TranslateServiceServer) alterTranslations(
+	ctx context.Context,
+	allMessages model.MessagesSlice,
+	newOriginalMessages *model.Messages,
+) (model.MessagesSlice, error) {
```

Comment:
Let the caller replace original Messages in allMessages and pass it to `alterTranslations()`.

```suggestion
func (t *TranslateServiceServer) alterTranslations(
	ctx context.Context,
	allMessages model.MessagesSlice,
) (model.MessagesSlice, error) {
```